src/eddymotion/dmri.py

```diff
@@ -211,35 +199,51 @@ def from_filename(cls, filename):
 
 
 def load(
-    filename, gradients_file=None, b0_file=None, brainmask_file=None, fmap_file=None
+    filename,
+    gradients_file=None,
+    b0_file=None,
+    brainmask_file=None,
+    fmap_file=None,
+    bvec_file=None,
+    bval_file=None,
+    b0_thres=50,
 ):
     """Load DWI data."""
     filename = Path(filename)
     if filename.name.endswith(".h5"):
         return DWI.from_filename(filename)
 
-    if not gradients_file:
+    if gradients_file:
+        grad = np.loadtxt(gradients_file, dtype="float32").T
+    elif bvec_file and bval_file:
+        grad = np.vstack(
+            (
+                np.loadtxt(bvec_file, dtype="float32"),
+                np.loadtxt(bval_file, dtype="float32"),
+            )
+        )
+    else:
         raise RuntimeError("A gradients file is necessary")
 
-    img = nb.as_closest_canonical(nb.load(filename))
-    retval = DWI(
-        affine=img.affine,
-    )
-    grad = np.loadtxt(gradients_file, dtype="float32").T
-    gradmsk = grad[-1] > 50
+    img = nb.load(filename)
+    fulldata = img.get_fdata(dtype="float32")
+    retval = DWI(affine=img.affine,)
+    gradmsk = grad[-1] > b0_thres
     retval.gradients = grad[..., gradmsk]
-    retval.dataobj = img.get_fdata(dtype="float32")[..., gradmsk]
+    retval.dataobj = fulldata[..., gradmsk]
 
     if b0_file:
-        b0img = nb.as_closest_canonical(nb.load(b0_file))
+        b0img = nb.load(b0_file)
         retval.bzero = np.asanyarray(b0img.dataobj)
+    elif np.any(~gradmsk):
+        retval.bzero = np.median(fulldata[..., ~gradmsk], axis=3)
```

Comment:
I've never seen anyone use median values as the b0 image before. Is that common?

Reply:
the median is more robust against outliers, but otherwise, it should be reasonably close to the average.

Reply:
The idea behind this is to have a fallback to generate a b0 quickly. The assumption is that generally, you will provide the `b0_file` argument with a b=0 calculated in a more sophisticated way (e.g., denoising, b1 drift compensation, etc.)